applications/RANSApplication/custom_strategies/algebraic_flux_corrected_scalar_steady_scheme.h

```diff
@@ -0,0 +1,567 @@
+
+        KRATOS_CATCH("")
+    }
+
+    void Update(ModelPart& rModelPart,
+                DofsArrayType& rDofSet,
+                TSystemMatrixType& rA,
+                TSystemVectorType& rDx,
+                TSystemVectorType& rb) override
+    {
+        KRATOS_TRY;
+
+        mpDofUpdater->UpdateDofs(rDofSet, rDx, mRelaxationFactor);
+
+        KRATOS_CATCH("");
+    }
+
+    void Clear() override
+    {
+        this->mpDofUpdater->Clear();
+    }
+
+    void CalculateSystemContributions(Element& rElement,
+                                      LocalSystemMatrixType& rLHS_Contribution,
+                                      LocalSystemVectorType& rRHS_Contribution,
+                                      Element::EquationIdVectorType& rEquationIdVector,
+                                      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+
+        const auto k = OpenMPUtils::ThisThread();
+
+        this->CalculateSystemMatrix<Element>(rElement, rLHS_Contribution, rRHS_Contribution,
+                                             mAuxMatrix[k], rCurrentProcessInfo);
+        rElement.EquationIdVector(rEquationIdVector, rCurrentProcessInfo);
+
+        this->CalculateArtificialDiffusionMatrix(mAuxMatrix[k], rLHS_Contribution);
+
+        AddAntiDiffusiveFluxes(rRHS_Contribution, rLHS_Contribution, rElement,
+                               mAuxMatrix[k]);
+        noalias(rLHS_Contribution) += mAuxMatrix[k];
+
+        rElement.GetValuesVector(mValues[k]);
+        noalias(rRHS_Contribution) -= prod(rLHS_Contribution, mValues[k]);
+
+        KRATOS_CATCH("");
+    }
+
+    void CalculateSystemContributions(Condition& rCondition,
+                                      LocalSystemMatrixType& rLHS_Contribution,
+                                      LocalSystemVectorType& rRHS_Contribution,
+                                      Element::EquationIdVectorType& rEquationIdVector,
+                                      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+
+        const auto k = OpenMPUtils::ThisThread();
+
+        this->CalculateSystemMatrix<Condition>(rCondition, rLHS_Contribution, rRHS_Contribution,
+                                               mAuxMatrix[k], rCurrentProcessInfo);
+        rCondition.EquationIdVector(rEquationIdVector, rCurrentProcessInfo);
+
+        KRATOS_CATCH("");
+    }
+
+    void CalculateRHSContribution(Element& rElement,
+                                  LocalSystemVectorType& rRHS_Contribution,
+                                  Element::EquationIdVectorType& rEquationIdVector,
+                                  const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+
+        const auto k = OpenMPUtils::ThisThread();
+        CalculateSystemContributions(rElement, mAuxMatrix[k], rRHS_Contribution,
+                                     rEquationIdVector, rCurrentProcessInfo);
```

Comment:
You are calculating the LHS within the RHS method calculation. Doesn't this affect the performance?

Reply:
RHS always require LHS ryt since our RHS is the residual?